include/cupla/device/math/Common.hpp

```diff
@@ -62,60 +66,59 @@ namespace detail
             >::value,
             "The math concept implementation for the type 'Acc' and 'AccThreadSeq' must be equal"
         );
+#endif
         return AccMathConcept{};
     }
 } // namespace detail
 
-/* Using the free alpaka functions `alpaka::math::*` will result into `__host__ __device__`
- * errors, therefore the alpaka math trait must be used.
- */
-#define CUPLA_UNARY_MATH_FN(functionName, alpakaMathConcept, alpakaMathTrait)  \
+#define CUPLA_UNARY_MATH_FN_DETAIL(accOrMathImpl, functionName, alpakaMathConcept, alpakaMathTrait)  \
```

Comment:
(Not important) Perhaps putting `accOrMathImpl` as a second argument would be a little more natural.

Reply:
I will keep it because the accelerator is always the first argument.

Never the less to understand you thinking, could you please explain why it is more natural to have `accOrMathImpl` as second parameter.

Reply:
Sure, I can also see your logic now. My logic was that first this macro creates a function and so the function name is the most logical first parameter, e.g. imagine a theoretical C++ counterpart `void declareFunction( functionName, ... );`. Secondly, to be more consistent with the previous implementation: `accOrMathImpl` kind of goes together with `alpakaMathConcept`, both logically and in the implementation, so I would put them next to each other as a "group".

Reply:
@sbastrakov thanks for the explanation. I will change the order to your suggestion. I have not seen it a factory/constructor function but it is.